src/sentry/sentry_apps/utils/errors.py

```diff
@@ -7,43 +8,41 @@ class SentryAppErrorType(Enum):
     SENTRY = "sentry"
 
 
-# Represents a user/client error that occured during a Sentry App process
-class SentryAppError(Exception):
-    error_type = SentryAppErrorType.CLIENT
-    status_code = 400
+class ErrorContext(TypedDict, total=False):
+    # Info that gets sent only to the integrator via webhook
+    webhook_context: dict[str, Any]
+    # Info that gets sent to the end user via endpoint Response AND sent to integrator
+    public_context: dict[str, Any]
+
+
+class SentryAppBaseError(Exception):
+    error_type: SentryAppErrorType
+    status_code: int
 
     def __init__(
         self,
-        error: Exception | None = None,
+        message: str,
         status_code: int | None = None,
+        extras: ErrorContext | None = None,
     ) -> None:
-        if status_code:
-            self.status_code = status_code
+        self.status_code = status_code or self.status_code
+        self.extras = extras or {}
+        self.message = message
+
+
+# Represents a user/client error that occured during a Sentry App process
+class SentryAppError(SentryAppBaseError):
+    error_type = SentryAppErrorType.CLIENT
```

Comment:
are you using `error_type` anywhere?

Reply:
we need error_type in the future for Alert Rule Action UI process/ creation because it's an RPC method and we need to send error context in the RPC response.